adapters/repos/db/lsmkv/segment_group.go

```diff
@@ -200,7 +200,7 @@ func newSegmentGroup(logger logrus.FieldLogger, metrics *Metrics,
 				return nil, fmt.Errorf("close already compacted right segment %s: %w", rightSegmentFilename, err)
 			}
 
-			err = rightSegment.drop()
+			err = rightSegment.dropImmediately()
```

Comment:
**question:** why do we keep the dropImmediatly here but do it delayed later? We should detail this in the code.

Reply:
Fair question, let me add a comment.

Reply:
Done, see next commit.